projects/dashboards/src/widgets/container/container-widget-renderer.component.ts

```diff
@@ -18,15 +20,17 @@ import { ContainerWidgetModel } from './container-widget.model';
     </div>
   `
 })
-export class ContainerWidgetRendererComponent extends WidgetRenderer<ContainerWidgetModel> implements AfterViewInit {
+export class ContainerWidgetRendererComponent
+  extends WidgetRenderer<ContainerWidgetModel | ConditionalContainerWidgetModel>
```

Comment:
So two issues with this approach that I see -
1. The renderer now has to know about different types of models (not a huge deal)
2. The conditional works at a container level - this seems more problematic. If I want to condition one widget (which is more likely the case), does that mean it gets padded? What if the condition should change the layout? In my mind, I was thinking the conditional would always work with a single widget - that widget however could be a container itself, and handle its own layout.

I think one nice to have thing with the single widget approach is that if we do eventually add support for conditional models, it would use the exact same model.

Reply:
Yeah, I had the same thoughts but actually stopped and stashed the new `conditional-container-widget.model.ts` and `conditional-container-widget-renderer.component.ts` I was going to with to get feedback on this first.

Is that the suggestion then? We want this to have its own model widget model and renderer?

Reply:
That'd be my suggestion, but of course alway welcome to push back. I would expect it to be no more code than this change (or close enough) and touch fewer existing files. It's trading a new pass through renderer for the changes to the existing renderer/container widget models. I also like the idea of going with simply `conditional.model.ts`- We can implement that by associating a renderer right now, but once we build out model support any JSON using this for widgets remains identical (because a widget is jut a model with an associated renderer)